Aardvark/ARKEmailBugReporter.m

```diff
@@ -652,3 +619,77 @@ - (BOOL)canBecomeFirstResponder;
+     Transfer first responder to an invisble view when a debug screenshot is captured to make bug filing itself bug-free.
+     */
+    [self _stealFirstResponder];
+    
+    NSString * const title = NSLocalizedString(@"What Went Wrong?", @"Title text for alert asking user to describe a bug they just encountered");
+    NSString * const message = NSLocalizedString(@"Please briefly summarize the issue you just encountered. You’ll be asked for more details later.", @"Subtitle text for alert asking user to describe a bug they just encountered");
+    NSString * const composeReportButtonTitle = NSLocalizedString(@"Compose Report", @"Button title to compose bug report");
+    NSString * const cancelButtonTitle = NSLocalizedString(@"Cancel", @"Button title to not compose a bug report");
+    
+    UIAlertController *const alertController = [UIAlertController alertControllerWithTitle:title message:message preferredStyle:UIAlertControllerStyleAlert];
+    
+    [alertController addAction:[UIAlertAction actionWithTitle:composeReportButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
+        UITextField *textfield = [alertController.textFields firstObject];
+        [self _createBugReportWithTitle:textfield.text];
+    }]];
+    
+    [alertController addAction:[UIAlertAction actionWithTitle:cancelButtonTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
+        self.completion(nil);
+    }]];
+    
+    [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
+        [self _configureAlertTextfield:textField];
+    }];
+    
+    UIViewController *viewControllerToPresentAlertController = [UIApplication sharedApplication].keyWindow.rootViewController;
+    while (viewControllerToPresentAlertController.presentedViewController != nil) {
+        viewControllerToPresentAlertController = viewControllerToPresentAlertController.presentedViewController;
+    }
+    
+    /*
+     Disabling animations here to avoid potential crashes resulting from unexpected view state in UIKit
+     */
+    [viewControllerToPresentAlertController presentViewController:alertController animated:NO completion:NULL];
+}
+
+- (void)_stealFirstResponder;
+{
+    ARKInvisibleView *invisibleView = [ARKInvisibleView new];
+    invisibleView.layer.opacity = 0.0;
+    [[UIApplication sharedApplication].keyWindow addSubview:invisibleView];
+    [invisibleView becomeFirstResponder];
+    [invisibleView removeFromSuperview];
+}
+
+- (void)_configureAlertTextfield:(UITextField *)textField
+{
+    textField.autocapitalizationType = UITextAutocapitalizationTypeSentences;
+    textField.autocorrectionType = UITextAutocorrectionTypeYes;
+    textField.spellCheckingType = UITextSpellCheckingTypeYes;
+    textField.returnKeyType = UIReturnKeyDone;
+}
+
+- (void)_createBugReportWithTitle:(NSString *)title;
+{
+    self.configuration.prefilledEmailSubject = title;
+    self.completion(self.configuration);
```

Comment:
should we set `completion` to `nil` after this line to prevent accidentally calling it twice?

Reply:
This is a short-lived object, so I don't think there's any change in calling it twice.  No harm in setting it to `nil` though, so probably better to be safe.